packages/devextreme/js/__internal/scheduler/appointments/data_provider/utils/is_appointment_matched_intervals.ts

```diff
@@ -0,0 +1,20 @@
+import type { DateInterval } from './type';
+
+export const isAppointmentMatchedIntervals = (
+  appointment: {
+    startDate: Date;
+    endDate: Date;
+  },
+  intervals: DateInterval[],
+): boolean => {
+  const { startDate, endDate } = appointment;
+  let i = 0;
+
+  for (; i < intervals.length; i += 1) {
+    if (startDate < intervals[i].max) {
+      break;
+    }
+  }
+
+  return i < intervals.length && (startDate >= intervals[i].min || endDate > intervals[i].min);
+};
```

Comment:
Let's refactor this function and make it more obvious

```suggestion
export const isAppointmentMatchedIntervals = (
  { startDate, endDate }: { startDate: Date; endDate: Date },
  intervals: DateInterval[],
): boolean => {
  for (let idx = 0; idx < intervals.length; idx += 1) {
    const { min, max } = intervals[idx];

    if (startDate < max && (startDate >= min || endDate > min)) {
      return true;
    }
  }

  return false;
};
```